website/content/docs/secrets/kubernetes.mdx

```diff
@@ -0,0 +1,339 @@
+   [Kubernetes RBAC documentation](https://kubernetes.io/docs/reference/access-authn-authz/rbac/).
+
+2. If Vault will not be automatically managing roles or service accounts (see
+   [Automatically Managing Roles and Service Accounts](#automatically-managing-roles-and-service-accounts)),
+   then we will need to set up a service account that Vault will issue tokens for.
+
+   ~> **Note**: It is highly recommended that the service account that Vault issues tokens for
+      is **NOT** the same service account that Vault itself uses.
+
+   We will run everything under the namespace `test`, which we can create if it does not already exist.
+
+   ```shell-session
+   $ kubectl create namespace test
+   namespace/test created
+   ```
+
+   Here is a simple set up of a service account, role, and role binding in the Kubernetes `test`
+   namespace with basic permissions we will use for this document:
+   ```yaml
+   apiVersion: v1
+   kind: ServiceAccount
+   metadata:
+     name: test-service-account-with-generated-token
+     namespace: test
+   ---
+   apiVersion: rbac.authorization.k8s.io/v1
+   kind: Role
+   metadata:
+     name: test-role-list-pods
+     namespace: test
+   rules:
+   - apiGroups: [""]
+     resources: ["pods"]
+     verbs: ["list"]
+   ---
+   apiVersion: rbac.authorization.k8s.io/v1
+   kind: RoleBinding
+   metadata:
+     name: test-role-abilities
+     namespace: test
+   roleRef:
+     apiGroup: rbac.authorization.k8s.io
+     kind: Role
+     name: test-role-list-pods
+   subjects:
+   - kind: ServiceAccount
+     name: test-service-account-with-generated-token
+     namespace: test
+   ```
+
+   We can create these objects with `kubectl apply -f`.
+
+3. Enable the Kubernetes Secrets Engine:
+
+   ```shell-session
+   $ vault secrets enable kubernetes
+   Success! Enabled the kubernetes Secrets Engine at: kubernetes/
+   ```
+
+   By default, the secrets engine will mount at the same name as the engine, i.e.,
+   `kubernetes/` here. This can be changed by passing the `-path` argument when enabling.
+
+4. Configure the mount point. An empty config is allowed.
+
+   ```shell-session
+   $ vault write -f kubernetes/config
+   ```
+
+   Configuration options are available as specified in the
+   [API docs](/api-docs/secret/kubernetes).
+
+5. We can now configure Kubernetes Secrets Engine to create a Vault role (**not** the same as a
+   Kubernetes role) that can generate service account tokens for the given service account:
+
+   ```shell-session
+   $ vault write kubernetes/roles/my-role \
+       allowed_kubernetes_namespaces="*" \
+       service_account_name="test-service-account-with-generated-token" \
+       token_default_ttl="10m"
+   ```
+
+## Generating Credentials
+
+After a user has authenticated to Vault and has sufficient permissions, a read request to the
+`creds` endpoint for the Vault role will generate and return a new service account token.
+
+```shell-session
+$ vault write kubernetes/creds/my-role \
```

Comment:
```suggestion
$ vault read kubernetes/creds/my-role \
```

Reply:
I think we were having some discussion about this, and agreed that `write` is better?

Reply:
Yep 👍